### PR TITLE
feat(route): add UNTAG分类 & 专题

### LINKED ISSUE #12570: Untag 文章抓取
### Category

New media

### Website URL

https://utgd.net

### Website description

前少数派编者创建的网站. 关注生产力工具.

这个网站的创作团队写作的主要出发点之一便是解决问题。故而以读者群的形式来运营。读者有问题，可进群提问 ，编者对自己产生的内容长期负责。如果看了文章，希望交流背后的思想；或者想把某个方法结合到自己的生活之中来实践，但一些隐私细节不便公开讨论甚至不想留下文字，皆可提出预约，通过语音与主笔交流，一对一解决问题。

### What content should be included?

- [方法](https://utgd.net/category/s/method)
- [观点](https://utgd.net/category/s/opinion)
- [专题](https://utgd.net/topic)

### Additional description

网站提供 订阅源 https://rss.utgd.net/feed/. 但是用它无法获取全文. 在 FreshRSS, MiniFlux 等软件中设置抓取的 `div class` 仍然无济于事. 特此求助于 RSSHub 社区

### This is not a duplicated rss request

- [X] I have searched [existing issues](https://github.com/DIYgod/RSSHub/issues) and [pull requests](https://github.com/DIYgod/RSSHub/pulls) to ensure this rss proposal has not already been requested

### PR DESCRIPTION
**Pull request by nczitzk**
## 该 PR 相关 Issue / Involved Issue

Close #12570

## 路由地址示例 / Example for the Proposed Route(s)

```routes
/utgd/method
/utgd/topic/在线阅读专栏
```

## 新 RSS 路由检查表 / New RSS Route Checklist
  
- [x] 新的路由 New Route
  - [x] 跟随 [v2 路由规范](https://docs.rsshub.app/joinus/script-standard.html) Follows [v2 Script Standard](https://docs.rsshub.app/en/joinus/script-standard.html)
- [x] 文档说明 Documentation
  - [x] 中文文档 CN
  - [ ] 英文文档 EN
- [x] 全文获取 fulltext
  - [x] 使用缓存 Use Cache
- [ ] 反爬/频率限制 anti-bot or rate limit?
  - [ ] 如果有, 是否有对应的措施? If yes, do your code reflect this sign?
- [x] [日期和时间](https://docs.rsshub.app/joinus/pub-date.html) [date and time](https://docs.rsshub.app/en/joinus/pub-date.html)
  - [x] 可以解析 Parsed
  - [x] 时区调整 Correct TimeZone
- [ ] 添加了新的包 New package added
- [ ] `Puppeteer`

### CODE REVIEW

File: lib/v2/untag/category.js [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+const path = require('path');
+
+module.exports = async (ctx) => {
+    const category = ctx.params.category ?? 'method';
+    const limit = ctx.query.limit ? parseInt(ctx.query.limit) : 20;
+
+    const rootUrl = 'https://utgd.net';
+    const apiUrl = `${rootUrl}/api/v2/pages`;
+    const currentUrl = `${rootUrl}/category/s/${category}`;
+    const slugUrl = `${rootUrl}/api/v2/category/slug/${category}`;
```
Comment (TonyRL) [inline lib/v2/untag/category.js:14]:
```suggestion
    const apiUrl = `${rootUrl}/api/v2/pages/`;
    const currentUrl = `${rootUrl}/category/s/${category}`;
    const slugUrl = `${rootUrl}/api/v2/category/slug/${category}/`;
```
https://github.com/DIYgod/RSSHub/actions/runs/5105246772/jobs/9176711212#step:12:17
https://github.com/DIYgod/RSSHub/actions/runs/5105246772/jobs/9176711212#step:12:19

File: lib/v2/untag/category.js [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+            limit,
+        },
+    });
+
+    const items = await Promise.all(
+        response.data.items.map((item) =>
+            ctx.cache.tryGet(`untag-${item.id}`, async () => {
+                const authorResponse = await got({
+                    method: 'get',
+                    url: `${rootUrl}/api/v2/user/profile/${item.article_author.id}`,
```
Comment (TonyRL) [inline lib/v2/untag/category.js:40]:
```suggestion
                    url: `${rootUrl}/api/v2/user/profile/${item.article_author.id}/`,
```
https://github.com/DIYgod/RSSHub/actions/runs/5105246772/jobs/9176711212#step:12:40

File: lib/v2/untag/category.js [hunk 3]
```diff
@@ -0,0 +1,65 @@
...
+                });
+
+                return {
+                    title: item.title,
+                    link: `${rootUrl}/article/${item.id}`,
+                    description: art(path.join(__dirname, 'templates/description.art'), {
+                        image: item.article_image,
+                        description: item.article_content,
```
Comment (TonyRL) [inline lib/v2/untag/category.js:48]: `item.article_content` includes Markdown content. You can use `markdown-it` to render it first.

File: lib/v2/untag/maintainer.js [hunk 4]
```diff
@@ -0,0 +1,3 @@
+module.exports = {
+    '/category/:category?': ['nczitzk'],
```
Comment (TonyRL) [inline lib/v2/untag/maintainer.js:2]:
```suggestion
    '/category/:category?': ['nczitzk'],
    '/topic/:topic?': ['nczitzk'],
```

File: lib/v2/untag/topic.js [hunk 5]
```diff
@@ -0,0 +1,67 @@
...
+const { art } = require('@/utils/render');
+const path = require('path');
+
+module.exports = async (ctx) => {
+    const topic = ctx.params.topic ?? '在线阅读专栏';
+    const limit = ctx.query.limit ? parseInt(ctx.query.limit) : 20;
+
+    const rootUrl = 'https://utgd.net';
+    const currentUrl = `${rootUrl}/topic`;
+    const topicUrl = `${rootUrl}/api/v2/topic`;
```
Comment (TonyRL) [inline lib/v2/untag/topic.js:13]:
```suggestion
    const topicUrl = `${rootUrl}/api/v2/topic/`;
```
https://github.com/DIYgod/RSSHub/actions/runs/5105246772/jobs/9176711212#step:12:62

File: lib/v2/untag/topic.js [hunk 6]
```diff
@@ -0,0 +1,67 @@
...
+
+    const topicItems = response.data.filter((i) => i.title === topic);
+
+    if (!topicItems) {
+        throw Error(`No topic named ${topic}`);
+    }
+
+    const topicItem = topicItems[0];
+
+    const apiUrl = `${rootUrl}/api/v2/topic/${topicItem.id}/article`;
```
Comment (TonyRL) [inline lib/v2/untag/topic.js:28]:
```suggestion
    const apiUrl = `${rootUrl}/api/v2/topic/${topicItem.id}/article/`;
```
https://github.com/DIYgod/RSSHub/actions/runs/5105246772/jobs/9176711212#step:12:64

File: lib/v2/untag/topic.js [hunk 7]
```diff
@@ -0,0 +1,67 @@
...
+        method: 'get',
+        url: apiUrl,
+    });
+
+    const items = await Promise.all(
+        response.data.slice(0, limit).map((item) =>
+            ctx.cache.tryGet(`untag-${item.id}`, async () => {
+                const detailResponse = await got({
+                    method: 'get',
+                    url: `${rootUrl}/api/v2/pages/${item.id}`,
```
Comment (TonyRL) [inline lib/v2/untag/topic.js:40]:
```suggestion
                    url: `${rootUrl}/api/v2/pages/${item.id}/`,
```
https://github.com/DIYgod/RSSHub/actions/runs/5105246772/jobs/9176711212#step:12:76

File: lib/v2/untag/topic.js [hunk 8]
```diff
@@ -0,0 +1,67 @@
...
+                        fields: 'article_content',
+                    },
+                });
+
+                return {
+                    title: item.title,
+                    link: `${rootUrl}/article/${item.id}`,
+                    description: art(path.join(__dirname, 'templates/description.art'), {
+                        image: item.article_image,
+                        description: detailResponse.data.article_content,
```
Comment (TonyRL) [inline lib/v2/untag/topic.js:51]: `detailResponse.data.article_content` includes Markdown content. You can use `markdown-it` to render it first.

File: docs/new-media.md [hunk 9]
```diff
@@ -1469,6 +1469,29 @@ Provides all of the Thrillist articles with the specified tag.
 
 <Route author="Rongronggg9" example="/tribalfootball" path="/tribalfootball" />
 
+## UNTAG
+
+### 分类
+
+<Route author="nczitzk" example="/untag/category/method" path="/untag/category/:category?" :paramsDesc="['分类，可在对应分类页的 URL 中找到，默认为方法']">
```
Comment (TonyRL) [inline docs/new-media.md:1476]: Please update route path to `/utgd/...`

File: lib/v2/untag/category.js (hunk 1 above)
Comment (nczitzk) [reply lib/v2/untag/category.js:14]: 👌

**Comment by KawausoJyou** [comment]
你好，是否可以增加不区分分类，直接获取首页最新文章的路由？
谢谢

**Pull request merged**